packages/upset/src/components/SizeBar.tsx

```diff
@@ -51,6 +46,22 @@ export const SizeBar: FC<Props> = ({ row, size }) => {
     rectArray[i] = i;
   }
 
+  function getFillColor(index: number) {
+    // if the row is bookmarked, highlight the bar with the bookmark color
+    if (row !== undefined && bookmarkedIntersections.some((bookmark) => bookmark.id === row.id)) {
+      // darken the color for advanced scale sub-bars
+      if (index !== 0) {
+        return newShade(bookmarkedColorPallete[row.id], -(12 + (index * 2)));
+      }
+      return bookmarkedColorPallete[row.id];
+    }
+
+    if (row !== undefined && currentIntersection !== null && currentIntersection.id === row.id) { // if currently selected, use the highlight colors
+      return highlightColors[index];
+    }
+    return colors[index];
```

Comment:
When do we get to line 62? Seems like the above stuff should be complete, either it's bookmarked or highlighted, right?

Reply:
`colors[index]` is for rows which are not being hovered, clicked, or bookmarked. i.e. every other row